python/ray/tests/test_multi_node_3.py

```diff
@@ -145,6 +144,47 @@ def test_calling_start_ray_head(call_ray_stop_only):
     assert blocked.returncode != 0, "ray start shouldn't return 0 on bad exit"
 
 
+def test_ray_start_non_head(call_ray_stop_only):
+
+    # Test that we can call ray start to connect to an existing cluster.
+
+    # Test starting Ray with a port specified.
+    check_call_ray(
+        ["start", "--head", "--port", "7298", "--resources", "{\"res_0\": 1}"])
+
+    # Test starting node connecting to the above cluster.
+    check_call_ray([
+        "start", "--address", "127.0.0.1:7298", "--resources", "{\"res_1\": 1}"
+    ])
+
+    # Test starting Ray with address `auto`.
+    check_call_ray(
+        ["start", "--address", "auto", "--resources", "{\"res_2\": 1}"])
+
+    # Run tasks to verify nodes with custom resources are available.
+    driver_script = """
+import ray
+ray.init()
+
+@ray.remote
+def f():
+    return 1
+
+assert ray.get(f.remote()) == 1
+assert ray.get(f.options(resources={"res_0": 1}).remote()) == 1
+assert ray.get(f.options(resources={"res_1": 1}).remote()) == 1
+assert ray.get(f.options(resources={"res_2": 1}).remote()) == 1
+print("success")
+"""
+    os.environ["RAY_ADDRESS"] = "auto"
+    out = run_string_as_driver(driver_script)
+    # Make sure the driver succeeded.
+    assert "success" in out
+
+    del os.environ["RAY_ADDRESS"]
```

Comment:
Better use monkey patch (https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/tests/test_scheduling.py?L494:14) to avoid pollution for the rest tests.